Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitFacadeHandler.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            if (!MixedRealityToolkit.IsInitialized)
+            {
+                return;
+            }
+
+            // When the play state changes just nuke everything and start over
+            DestroyAllChildren();
+        }
+
+        private static void UpdateServiceFacades(SceneView sceneView)
+        {
+            // If we're not using inspectors, destroy them all now
+            if (!MixedRealityToolkit.IsInitialized || MixedRealityToolkit.Instance.HasActiveProfile && !MixedRealityToolkit.Instance.ActiveProfile.UseServiceInspectors)
+            {
+                DestroyAllChildren();
+                return;
+            }
+
+            servicesToSort.Clear();
+
+            foreach (IMixedRealityService service in MixedRealityToolkit.Instance.ActiveSystems.Values)
+            {
+                servicesToSort.Add(service);
+            }
+
+            foreach (Tuple<Type, IMixedRealityService> registeredService in MixedRealityToolkit.Instance.RegisteredMixedRealityServices)
+            {
+                servicesToSort.Add(registeredService.Item2);
+            }
+
+            servicesToSort.Sort(
+                delegate (IMixedRealityService s1, IMixedRealityService s2)
+                {
+                    string s1Name = s1.GetType().Name;
+                    string s2Name = s2.GetType().Name;
+
+                    if (s1Name == s2Name)
+                    {
+                        return s1.Priority.CompareTo(s2.Priority);
+                    }
+
+                    return s1Name.CompareTo(s2Name);
+                });
+
+            for (int i = 0; i < servicesToSort.Count; i++)
+            {
+                CreateFacade(MixedRealityToolkit.Instance.transform, servicesToSort[i], i);
+            }
+
+            // Delete any stragglers
+            childrenToDelete.Clear();
+            for (int i = servicesToSort.Count; i < MixedRealityToolkit.Instance.transform.childCount; i++)
+            {
+                childrenToDelete.Add(MixedRealityToolkit.Instance.transform.GetChild(i));
+            }
+
+            foreach (Transform childToDelete in childrenToDelete)
+            {
+                if (Application.isPlaying)
+                {
+                    GameObject.Destroy(childToDelete.gameObject);
+                }
+                else
+                {
+                    GameObject.DestroyImmediate(childToDelete.gameObject);
+                }
+            }
+
+            try
+            {
+                // Update all self-registered facades
+                foreach (ServiceFacade facade in ServiceFacade.ActiveFacadeObjects)
+                {
+                    if (facade == null)
+                    {
+                        continue;
+                    }
+
+                    facade.CheckIfStillValid();
+                }
+            }
+            catch(Exception e)
+            {
+                e = null;
```

Comment:
These lines generate a compiler warning in unity (i.e. e being unused):

Assets\MixedRealityToolkit\Inspectors\MixedRealityToolkitFacadeHandler.cs(109,29): warning CS0168: The variable 'e' is declared but never used